File: c/anagram_checker.c
#include <stdbool.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

bool is_anagram(char *, char *);

int main(void) {
  char word1[30];
  char word2[30];

  puts("Enter 2 strings to determine if they are anagrams.");
  printf("Enter first string: ");
  scanf("%s", word1);

  printf("Enter second string: ");
  scanf("%s", word2);

  if (is_anagram(word1, word2)) {
    printf("\"%s\" and \"%s\" are anagrams.\n", word1, word2);
  } else {
    printf("\"%s\" and \"%s\" are not anagrams.\n", word1, word2);
  }

  return EXIT_SUCCESS;
}

bool is_anagram(char *word1, char *word2) {
  if (strlen(word1) != strlen(word2)) {
    return false;
  }

  for (int i = 0; i < (int)strlen(word2); i++) {
    if (word1[0] == word2[i]) {
      return true;
    }
  }

  return false;
}

File: zig/anagram_checker.zig
// FIX: a bug currently!!
const std = @import("std");

pub fn main() !void {
    const stdin = std.io.getStdIn().reader();
    var buffer: [1024]u8 = undefined;

    std.debug.print("Enter two strings and I'll tell you if they are anagrams:\n", .{});
    std.debug.print("Enter the first string: ", .{});
    var input = try stdin.readUntilDelimiter(buffer[0..], '\n');
    const first_string = input;

    std.debug.print("Enter the second string: ", .{});
    input = try stdin.readUntilDelimiter(buffer[0..], '\n');
    const second_string = input;

    const anagram = is_anagram(first_string, second_string);
    std.debug.print("{any}\n", .{anagram});
    if (anagram) {
        std.debug.print("\"{s}\" and \"{s}\" are anagrams", .{ first_string, second_string });
    } else {
        std.debug.print("\"{s}\" and \"{s}\" are not anagrams", .{ first_string, second_string });
    }
}

fn is_anagram(first: []const u8, second: []const u8) bool {
    if (first.len != second.len) return false;

    for (first, 0..) |value, idx| {
        if (value == second[idx]) return true;
    }

    return false;
}
